keps/sig-node/asier/README.md

```diff
@@ -0,0 +1,530 @@
+    - [Monitoring and alerting](#monitoring-and-alerting)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [GA](#ga)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+  - [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+
+<!-- /toc -->
+
+  
+
+## Release Signoff Checklist
+
+  
+
+<!--
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+
+of the targeted release**.
+
+  
+
+For enhancements that make changes to code or processes/procedures in core
+
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+
+Signoff checklist to be completed.
+
+  
+
+Check these off as they are completed for the Release Team to track. These
+
+checklist items _must_ be updated for the enhancement to be released.
+
+-->
+
+  
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+<!--
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+-->
+
+  
+
+[kubernetes.io]:  https://kubernetes.io/
+
+[kubernetes/enhancements]:  https://git.k8s.io/enhancements
+
+[kubernetes/kubernetes]:  https://git.k8s.io/kubernetes
+
+[kubernetes/website]:  https://git.k8s.io/website
+
+  
+
+## Summary
+
+  
+
+IMA namespaces allow to check the file integrity. This proposal adds file integrity inside containers deployed in kubernetes.
+
+  
+  
+
+## Motivation
+
+  
+
+File integrity is a way to improve security in systems allowing to:
+
+* Detect illicit activity
+
+* Detect unintended changes
+
+* Verify the status and health of the system
+
+* Comply with access rules
+
+  
+
+This can be achieved using IMA (Integrity Measurement Architecture) and EVM (Extended Verification Module), which can use a TPM chip as Hardware Root of Trust for high security environments.
+
+  
+
+### Goals
+
+  
+
+* Allow IMA to work inside a container with remote attestation
+
+  
+
+### Non-Goals
+
+  
+
+## Proposal
+  
+We propose to enable IMA linux namespaces in pods.
+
+Since IMA namespaces can be created when a container is launched, we can provide transparent integrity verification on any linux container.
+
+IMA and EVM can use a TPM chip as a hardware root of trust. Hence we can verify images against a set of golden hash values, as well as avoiding any further changes to the overlayfs to intercept calls and check the integrity of files.
+
+
+### User Stories (Optional)
```

Comment:
The current user-story seem more aspirational for me (maybe I'm missing something), that is why I posted some questions to them (to see if they are really aspirational or do we have a high-level idea on how that can be accomplished). If we don't really know how that can be accomplished and they seem mostly aspirational, let's just remove them for now. 

What actual things do you expect to be possible? I think this section should be that, not something aspirational. Something this will allow.

It is usually helpful to have this section completed early, but if you still don't know, leave it empty.

Maybe things that _can_ be done is something like this? No idea, though:

```
 As a cluster admin, I want to guarantee the rootfs files are unchanged and verified when a container access it, so I can have a high guarantee that files were not tampered.
```

Maybe more can be created after the measure mode, some others after the appraisal mode too (if we plan to support it)?

But let's keep this section concrete and real, or not fill them yet if we need to play a little more with the feature to understand what will be possible.

Reply:
Thanks for the tip. I will leave it empty for now.